src/sas/cli.py

```diff
@@ -0,0 +1,90 @@
+"""
+SasView command line interface.
+
+sasview -m module [args...]
+    Run module as main.
+sasview -c "python statements"
+    Execute python statements with sasview libraries available.
+sasview -i 
+    Start ipython interpreter.
+sasview script [args...]
+    Run script with sasview libraries available
+sasview
+    Start sasview gui
+"""
+import sys
+
+# TODO: Support dropping datafiles onto .exe?
+# TODO: Maybe use the bumps cli with project file as model?
+
+import argparse
+parser = argparse.ArgumentParser()
+parser.add_argument("-m", "--module", type=str,
+    help="Run module as main")
+parser.add_argument("-c", "--command", type=str,
+    help="Execute command")
+parser.add_argument("-i", "--interactive", action='store_true', 
+    help="Run interactive command line")
+parser.add_argument("argv", nargs="*",
+    help="script followed by argv")
+
+def exclusive_error():
+    print("Use only one of -m module args, -c command, -i, or script.py args.", sys.stderr)
+    sys.exit(1)
+
+def run_interactive():
+    """Run sasview as an interactive python interpreter"""
+    try:
+        from IPython import start_ipython
+        sys.argv = ["ipython", "--pylab"]
+        sys.exit(start_ipython())
+    except ImportError:
+        import code
+        code.interact(local={'exit': sys.exit})
+
+def main(logging="production"):
+    from sas.system import log
+    from sas.system import lib
+
+    # Eventually argument processing might affect logger or config, so do it first
+    args = parser.parse_args()
+
+    # Setup logger and sasmodels
+    if logging == "production":
+        log.production()
+    elif logging == "development":
+        log.development()
+    else:
+        raise ValueError(f"Unknown logging mode \"{logging}\"")
+    lib.setup_sasmodels()
+
+    # Parse mutually exclusive command line options
+    # mutually exclusive (-m module args, -c command, -i, script args)
+    if args.argv and not args.module: # script [arg...]
+        import runpy
+        if args.command or args.module or args.interactive:
+            exclusive_error()
+        sys.argv = args.argv
+        runpy.run_path(args.argv[0], run_name="__main__")
```

Comment:
Error handling from `runpy.run_*` functions is limited to exception handling. Raising an exception in the called code will rattle back and lead to this exiting non-zero. It's probably enough, although all failures will give the same exit code. If the function just returns `False` or `None` or any other common signal of failure, it will look like it has passed OK.

It probably needs to be documented that the custom script should raise an exception in the case of an error, not just return.

Reply:
How is this behaviour different from `python script.py`? Scripts don't return error codes unless they include `sys.exit(nonzero)`.